### ISSUE TITLE
SC2139 when using $HOME

### ISSUE THREAD

**Issue by fekir** [koalaman/shellcheck#829]
#### For bugs
- Rule Id: SC2139
- My shellcheck version: 0.4.4
- I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
-  I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env sh
alias tree_my_dir="tree $HOME/my_dir/"

```

#### Here's what shellcheck currently says:
SC2139: This expands when defined, not when used. Consider escaping.

#### Here's what I wanted or expected to see:
I don't think that the warning makes sense, since I would not expect $HOME to change, like for example $PWD. 
Maybe shellcheck could have a whitelist and not trigger the warning if the variable is $HOME, or are there some scenarios where this warning makes sense?

**Comment by kurahaupo** [comment]
The combination of "env" and "sh" is useless; "sh" is *always* /bin/sh on a
POSIX system.

And using "alias" in a POSIX shell script is broken. If "sh" isn't bash, it
probably won't work.

In fact, aliases inside scripts don't work how you think they should
either; the script may be parsed in its entirety before execution, and
aliases only take effect after they are executed.

Functions are also subject to the same "only after executed" rule, but they
don't affect how the script is parsed, only how the commands are
interpreted as they are executed.

**Comment by fekir** [comment]
Hi,thank you for your response and clarification

Maybe t is possible to add a new rule to shellcheck that issue a warning when using "anv" and "sh" instead of only "sh" ;-)

I'm using the alias in a file that is sourced, that defines some aliases and functions to use in the interactive shell.
As far as I know there aren't differences between scripts and files that are sourced.
I added "#!/usr/bin/env bash" (not sh) at the begin of the file in order to help shellcheck to understand against what rules it has to check (and indeed it helped a lot). 
The exact same warning appears also when using bash, so I thought there was no difference, sorry.

**Comment by koalaman** [comment]
You're right, this would not be an issue in practice because `$HOME` is generally a benign variable.

There's a good reason for still quoting variables that will never change though: you avoid double evaluation. I would seriously question this line, even though it's pretty much the same as yours:

    alias tree_my_dir='eval tree $HOME/my_dir/' 

On principle, it would probably be better to avoid this by default. I've added a note about this to the wiki. Thanks for reporting!

**Comment by fekir** [comment]
Sorry, I missed the notification.

So shellcheck will still trigger a warning in the new version? 
Or is it possible to whitelist "constants" variables? Maybe even only locally through a configuration file?
